lib/ignore/ignore.go

```diff
@@ -563,7 +564,12 @@ func parseIgnoreFile(fs fs.Filesystem, fd io.Reader, currentFile string, cd Chan
 		case strings.HasSuffix(line, "/**"):
 			err = addPattern(line)
 		case strings.HasSuffix(line, "/"):
-			err = addPattern(line + "**")
+			err = addPattern(line + "*")
+		// Used to widen the pattern accepted by ** in accordance to
+		// the specified behavior in syncthing.
+		case interDoubleStar.MatchString(line):
+			err = addPattern(interDoubleStar.ReplaceAllString(line, `$1*/**/*$2`))
+			err = addPattern(strings.ReplaceAll(line, "**", "*"))
```

Comment:
A check is required since `err` might be non-nil here, you can adopt the same pattern as in the `default:` case below.